glue/core/roi.py

```diff
@@ -163,21 +177,37 @@ class RectangularROI(Roi):
 
     """
     A 2D rectangular region of interest.
+
+    Parameters
+    ----------
+    xmin, xmax :  float, optional
+        x coordinates of left and right border
+    ymin, ymax :  float, optional
+        y coordinates of lower and upper border
+    theta : float, optional
+        Angle of anticlockwise rotation around center
     """
 
-    def __init__(self, xmin=None, xmax=None, ymin=None, ymax=None):
+    def __init__(self, xmin=None, xmax=None, ymin=None, ymax=None, theta=None):
         super(RectangularROI, self).__init__()
         self.xmin = xmin
         self.xmax = xmax
         self.ymin = ymin
         self.ymax = ymax
+        self.theta = 0 if theta is None else theta
+        if np.isclose(self.theta % np.pi, 0.0, atol=1e-9):
```

Comment:
Could this logic be inside the ``rotation`` function.

Reply:
Probably worth a shot, although after only calling `rotation_matrix_2d` as needed, the extra tests for exact multiples of right angles will still be needed in most places, to decide if the whole (much slower) process of pre-selecting and back-rotating points is to be initiated at all.

Reply:
Moved from `__init__` to `__contains__`, which is the caller for `rotation_matrix_2d`.